dd-sdk-android/src/main/kotlin/com/datadog/android/v2/core/DatadogCore.kt

```diff
@@ -183,14 +185,33 @@ internal class DatadogCore(
     }
 
     /** @inheritDoc */
-    override fun setFeatureContext(featureName: String, context: Map<String, Any?>) {
-        contextProvider?.setFeatureContext(featureName, context)
+    override fun updateFeatureContext(
+        featureName: String,
+        updateCallback: (context: MutableMap<String, Any?>) -> Unit
+    ) {
+        val feature = features[featureName] ?: return
+        contextProvider?.let {
+            // workaround for the backward compatibility with DatadogPlugin, we don't want to have
+            // context update in plugins in the synchronized block
+            val updatedContext = synchronized(feature) {
+                val featureContext = it.getFeatureContext(featureName)
+                val mutableContext = featureContext.toMutableMap()
+                updateCallback(mutableContext)
+                it.setFeatureContext(featureName, mutableContext)
+                mutableContext
+            }
+            if (featureName == RumFeature.RUM_FEATURE_NAME) {
```

Comment:
why are we doing this only for the `RUM` feature name ?

Reply:
Because context which is received by `DatadogPlugin` contains [only RUM-related info](https://github.com/DataDog/dd-sdk-android/blob/feature/sdkv2/dd-sdk-android/src/main/kotlin/com/datadog/android/plugin/DatadogContext.kt), so it makes sense to update it only when RUM context is updated.

Anyway, `DatadogPlugin` will be removed in v2. This code won't live long.